Use position/direction sampling from base source for radionuclide sources

ftessier opened the pull request:
This pull request replaces #672. The original pull request branch was removed from the forked repository, so the branch and pull request is hereby reinstated. The original pull request message was:

> The implementation of `egs_radionuclude_source` is changed so that it takes an input "base source" instead of defining a collimated or isotropic source. This way you can use any of the egs sources or a custom egs source to define the location of the radionuclide in the simulation. The change in particular was made towards fast simulations targeted radionuclide therapy across a whole patient phantom.

ftessier commented:
Rebased on develop, applied astyle.

rtownson commented:
Yes the missing contributor/documentation looks like a mistake, not sure how it happened.

When you re-instate it, change the indentations to use spaces instead of tabs.

ftessier commented:
Yes I will run astyle. So are you confirming that the other changes in that commit are all good then? I will attempt to clean it up and wait for review.

rtownson commented:
> Yes I will run astyle. So are you confirming that the other changes in that commit are all good then? I will attempt to clean it up and wait for review.

Yes, the comments from the first commit are correct. The code added in the second commit is also correct.

This PR breaks backwards compatibility - users with radionuclide sources will have to update their input files. I think that's OK, maintaining backwards compatibility would result in a lot of duplicate code.

rtownson commented:
I'm going to add one more commit to add a check for the old depreciated inputs, and give a description of the new format.

blakewalters commented on HEN_HOUSE/egs++/sources/egs_radionuclide_source/egs_radionuclide_source.h:
```diff
@@ -137,20 +137,19 @@ on the number of disintegration events (tracked by <b> \c ishower </b>). This
-source and target shapes.
+A radionuclide source is defined using the following input. It imports a base
+source and uses the base source getNextParticle() invocation to determine decay
+location. This implementation leads to increased random number sampling than
```
Why don't we just use an EGS_Shape instead of defining a base source if this is only used to define decay location?

rtownson replied:
Maybe the text is not quite clear - it also uses the direction sampling and getFluence from the base source, so it can uses the more complex source modelling (like a collimated source, which is not just a shape).

blakewalters replied:
Ah okay.  Yes, I see further up in the text it says it makes use of getFluence, but maybe it should also mention that it uses the incident directions as well.

rtownson replied:
Done, thanks!

blakewalters commented on HEN_HOUSE/egs++/sources/egs_radionuclide_source/egs_radionuclide_source.h:
```diff
@@ -255,9 +234,8 @@ results for non-disintegration emissions.
+        library             = egs_isotropic_source
         geometry            = my_envelope
         region selection    = IncludeSelected
         selected regions    = 1 2
```
Although I now see the base source (source_location), I don't see the egs_radiohead_source (i.e. my_source) defined in the updated example.  Is this missing?

rtownson replied:
You're right! I've fixed this.

blakewalters replied:
Thanks, @rtownson!

ftessier commented:
Rebased on develop, squashed and tweaked commit message. Also, remove the year from the contributor line in the header, to match EGSnrc style (only the author has the year indicated, to preserve pre-git history; now contribution date is provided by the git log)
